### PR TITLE
jupyter-server-proxy 3.2.0, metadata, deps, maintainer, tests

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes:
- syncs with upstream dependencies
- performs some CLI smoke tests
- includes the third-party licenses for vendored JS code (might be removed next release) 
- adds self as maintainer :wave:
- cleans up some template stuff

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by consideRatio** [comment]
This looks reasonable to me, but I didn't understand the changes so well.

**Review by consideRatio — APPROVED** [review]

**Comment by consideRatio** [comment]
@bollwyvl what did you do do get 91ec54b committed? Did you run some tools locally on your computer?

File: recipe/meta.yaml
Comment (consideRatio) [inline recipe/meta.yaml]: I think I'd be opined towards not hardcoding this like that, mainly because the `/j/` part in the url wasn't something I would have guessed should have been updated if jupyter-server-proxy would be renamed or similar perhaps.

I think this is fine though, I know too little about conda-forge practices and such to say what aligns itself well to make maintenance matters easy.

Comment (bollwyvl) [reply recipe/meta.yaml]: There's generally been a trend towards fewer jinja variables... In the case of a hard fork/rename, a new feedstock might well make the most sense, as have _even more_ naming dissonance doesn't help much.

Comment (jakirkham) [reply recipe/meta.yaml]: Yeah this kind of thing was more prevalent pre-[Grayskull]( https://github.com/conda-incubator/grayskull )

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml:27]: Should we hot-fix the old packages to have this constraint as well given it was [present in upstream since 3.0.0]( https://github.com/jupyterhub/jupyter-server-proxy/blob/4949a7eaf90c20a1c629e7f9420e8772464dc0d8/setup.py#L90 )?

**Comment by jakirkham** [comment]
> what did you do do get [91ec54b](https://github.com/conda-forge/jupyter-server-proxy-feedstock/commit/91ec54b5d45413acb5b0c96641d1ae332d89c68d) committed? Did you run some tools locally on your computer?

Yeah should be able to [re-render locally or with the bot]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ). Above the bot gives the same result ( https://github.com/conda-forge/jupyter-server-proxy-feedstock/pull/23#issuecomment-982178964 )

File: recipe/meta.yaml
Comment (consideRatio) [reply recipe/meta.yaml]: :tada: nice thanks for the clarification!

File: recipe/meta.yaml
Comment (bollwyvl) [reply recipe/meta.yaml:27]: is this desired before this one is merged?

**Comment by bollwyvl** [comment]
>  the bot gives the same result

Right, trust but verify.

Having `conda-smithy` around is handy, anyway, if only for the linter, e.g. `conda smithy recipe-lint recipe` in most feedstocks.

File: recipe/meta.yaml
Comment (jakirkham) [reply recipe/meta.yaml:27]: It can be done after

**Comment by maresb** [comment]
This looks excellent to me! We're just waiting for more reviewers to weigh in, right?

**Comment by consideRatio** [comment]
I'm not feeling very confident about conda-forge things so I hesitated on a merge initially, but given @jakirkham, @maresb, and @bollwyvl all think its looks good - I think we should go for a merge.

**Pull request merged**
